src/Explorer/Tabs/Tabs.tsx

```diff
@@ -220,7 +229,7 @@ const getReactTabContent = (activeReactTab: ReactTabKind, explorer: Explorer): J
     case ReactTabKind.Quickstart:
       return <QuickstartTab explorer={explorer} />;
     case ReactTabKind.QueryCopilot:
-      return <QueryCopilotTab initialInput={useTabs.getState().queryCopilotTabInitialInput} explorer={explorer} />;
+      return <QueryCopilotTab initialInput={useTabs.getState().queryCopilotTabInitialInput} explorer={explorer} isTabWorking={useTabs.getState().setIsTabExecuting} />;
```

Comment:
You don't have to pass the `setIsTabExecuting` function to the query copilot tab. You can directly call `useTabs.getState().setIsTabExecuting()` inside the tab by importing `useTabs`.

Reply:
Much more sensible, made the change.